analyzers/src/SonarAnalyzer.CSharp/Rules/DoNotOverwriteCollectionElements.cs

```diff
@@ -155,9 +152,9 @@ private static SyntaxNode GetIdentifier(ExpressionSyntax expression)
                     // ElementAccess    // <-- we are here
                     //   Arguments: index
                     //   Expression: SimpleMemberAccess
-                    //                 Expression: a
-                    //                 Name: b  // <-- we need this
-                    return ((MemberAccessExpressionSyntax)expression).Name;
+                    //                 Expression: a // <-- we need this
+                    //                 Name: b
```

Comment:
This comment is not correct. We don't need `a`. We need `a.b` and that matches the implementation.